scripts/updates/api/termux_github_api_get_tag.sh

```diff
@@ -99,62 +88,50 @@ termux_github_api_get_tag() {
 	# Why printf "%s\n"? Because echo interpolates control characters, which jq does not like.
 	response="$(printf "%s\n" "${response%|*}")"
 
-	local tag_name
-	if [[ "${http_code}" == "200" ]]; then
-		if [[ "${FILTER_REGEX}" ]]; then
-			if jq --exit-status --raw-output "${jq_filter}" <<<"${response}" >/dev/null; then
-				tag_name="$(jq --exit-status --raw-output "${jq_filter}" <<<"${response}" \
-					| sed 's/^v//' | grep -P "${FILTER_REGEX}" | head -n 1)"
-				if [[ -z "${tag_name}" ]]; then
-					termux_error_exit "No tags matched regex '${FILTER_REGEX}' in '${response}'"
+	local tag_name=""
+
+	case "${http_code}" in
+		200)
+			if tag_name="$(jq --exit-status --raw-output "${jq_filter}" <<<"${response}")"; then
+				tag_name="$(sed 's/^v//' <<<"${tag_name}")"
+				if [[ -n "${filter_regex}" ]]; then
+					tag_name="$(grep -P "${filter_regex}" <<<"${tag_name}" | head -n 1)"
+					[[ -z "${tag_name}" ]] && termux_error_exit "No tags matched regex '${filter_regex}' in '${response}'"
 				fi
 			else
 				termux_error_exit "Failed to parse tag name from: '${response}'"
 			fi
-		else
-			if jq --exit-status --raw-output "${jq_filter}" <<<"${response}" >/dev/null; then
-				tag_name="$(jq --exit-status --raw-output "${jq_filter}" <<<"${response}")"
+			;;
+		404)
+			if jq --exit-status "has(\"message\") and .message == \"Not Found\"" <<<"${response}"; then
+				termux_error_exit "No '${tag_type}' found (${api_url}). Try using other options."
 			else
-				termux_error_exit "Failed to parse tag name from: '${response}'"
+				termux_error_exit <<-EndOfError
+					ERROR: Failed to get '${tag_type}'(${api_url})'.
+					Response:
+					${response}
+				EndOfError
 			fi
-		fi
-	elif [[ "${http_code}" == "404" ]]; then
-		if jq --exit-status "has(\"message\") and .message == \"Not Found\"" <<<"${response}"; then
-			termux_error_exit <<-EndOfError
-				ERROR: No '${TAG_TYPE}' found (${api_url}).
-					Try using '$(
-					if [ "${TAG_TYPE}" = "newest-tag" ]; then
-						echo "latest-release-tag"
-					else
-						echo "newest-tag"
-					fi
-				)'.
-			EndOfError
-		else
+			;;
+		*)
 			termux_error_exit <<-EndOfError
-				ERROR: Failed to get '${TAG_TYPE}'(${api_url})'.
+				ERROR: Failed to get '${tag_type}'(${api_url})'.
+				HTTP code: ${http_code}
 				Response:
 				${response}
 			EndOfError
-		fi
-	else
-		termux_error_exit <<-EndOfError
-			ERROR: Failed to get '${TAG_TYPE}'(${api_url})'.
-			HTTP code: ${http_code}
-			Response:
-			${response}
-		EndOfError
-	fi
+		;;
+	esac
 
 	# If program control reached here and still no tag_name, then something went wrong.
-	if [[ -z "${tag_name:-}" ]] || [[ "${tag_name}" == "null" ]]; then
+	if [[ -z "${tag_name}" ]] || [[ "${tag_name}" == "null" ]]; then
 		termux_error_exit <<-EndOfError
-			ERROR: JQ could not find '${TAG_TYPE}'(${api_url})'.
+			ERROR: JQ could not find '${tag_type}'(${api_url})'.
 			Response:
 			${response}
 			Please report this as bug.
 		EndOfError
 	fi
```

Comment:
I've actually found that moving the tag validation to `termux_pkg_upgrade_version` simplified the Gitlab version significantly, and most of the validation was already present in that later function.